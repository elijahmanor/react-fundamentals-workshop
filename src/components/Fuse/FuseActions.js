import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import "./Fuse.css";

const FuseActions = ({ id, userName, bomb, onReply, onBomb }) => {
  const classes = classNames("fa fa-bomb", {
    "is-active": bomb
  });
  onBomb = onBomb.bind(null, { id, bomb: !bomb });
  return (
    <div className="FuseActions">
      <button className="FuseActions-action" onClick={onReply}>
        <i className="fa fa-comment-o" aria-hidden="true" />
      </button>
      <button className="FuseActions-action is-active" onClick={onBomb}>
        <i className={classes} aria-hidden="true" />
      </button>
    </div>
  );
};

FuseActions.propTypes = {
  id: PropTypes.number.isRequired,
  userName: PropTypes.string.isRequired,
  bomb: PropTypes.bool.isRequired,
  onReply: PropTypes.func,
  onBomb: PropTypes.func
};

FuseActions.defaultProps = {
  onReply: () => {},
  onBomb: () => {}
};

export default FuseActions;
