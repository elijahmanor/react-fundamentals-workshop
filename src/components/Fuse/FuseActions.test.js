import React from "react";
import { shallow, mount } from "enzyme";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import FuseActions from "./FuseActions";

describe("FuseActions", () => {
  beforeEach(() => {});

  it("renders unfavorited", () => {
    const tree = renderer.create(<FuseActions id={1} userName="johnsmith" favorite={false} />);
    expect(tree).toMatchSnapshot();
  });

  it("renders favorited", () => {
    const tree = renderer.create(<FuseActions id={1} userName="johnsmith" favorite={true} />);
    expect(tree).toMatchSnapshot();
  });
});
