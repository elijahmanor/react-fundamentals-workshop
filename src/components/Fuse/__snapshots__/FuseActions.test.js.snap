// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FuseActions renders favorited 1`] = `
<div
  className="FuseActions"
>
  <button
    className="FuseActions-action"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-comment-o"
    />
  </button>
  <button
    className="FuseActions-action is-active"
  >
    <i
      aria-hidden="true"
      className="fa fa-heart is-active"
    />
  </button>
</div>
`;

exports[`FuseActions renders unfavorited 1`] = `
<div
  className="FuseActions"
>
  <button
    className="FuseActions-action"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-comment-o"
    />
  </button>
  <button
    className="FuseActions-action is-active"
  >
    <i
      aria-hidden="true"
      className="fa fa-heart-o"
    />
  </button>
</div>
`;
